backend/remote/backend_state.go

```diff
@@ -79,7 +80,8 @@ func (r *remoteClient) Put(state []byte) error {
 	// Create the new state.
 	_, err = r.client.StateVersions.Create(ctx, r.workspace.ID, options)
 	if err != nil {
-		return fmt.Errorf("Error creating remote state: %v", err)
+		r.stateUploadErr = true
```

Comment:
Silly question: but if an instance of the remote client gets into an error state, how/where does it recover such that this is set to false/default, if that situation changes (or how would it?)

Reply:
Not a silly question! In code it will never recover, but that is fine as `terraform` is always executed for a single command (i.e. refresh, plan, apply. So if a certain command fails the binary will exit which then "resets" the remote client.

Does that make sense?

Reply:
I think I've got it, but checking: so a new instance of remoteClient is always created on a command, then?

Reply:
> So if a certain command fails the binary will exit which then "resets" the remote client.

As `r.stateUploadErr = true` is only set in memory, it will not persist when the binary (`terraform`) exits and the updated instance of `remoteClient` ceases to exist.

When you then execute a new command (e.g. `terraform plan`, `terraform apply`) the binary will start from scratch again which will create a new instance of `remoteClient`.

Reply:
Super helpful, thank you!